dartagnan/src/main/java/com/dat3m/dartagnan/program/processing/compilation/VisitorOpenCL.java

```diff
@@ -0,0 +1,137 @@
+package com.dat3m.dartagnan.program.processing.compilation;
+
+import com.dat3m.dartagnan.configuration.Arch;
+import com.dat3m.dartagnan.expression.Expression;
+import com.dat3m.dartagnan.expression.type.BooleanType;
+import com.dat3m.dartagnan.expression.Type;
+import com.dat3m.dartagnan.program.Register;
+import com.dat3m.dartagnan.program.event.EventFactory;
+import com.dat3m.dartagnan.program.event.Tag;
+import com.dat3m.dartagnan.program.event.Tag.C11;
+import com.dat3m.dartagnan.program.event.Event;
+import com.dat3m.dartagnan.program.event.MemoryEvent;
+import com.dat3m.dartagnan.program.event.arch.c11.OpenCLBarrier;
+import com.dat3m.dartagnan.program.event.core.*;
+import com.dat3m.dartagnan.program.event.core.RMWStore;
+import com.dat3m.dartagnan.program.event.lang.catomic.*;
+import com.dat3m.dartagnan.program.event.metadata.MemoryOrder;
+
+import java.util.List;
+
+import static com.dat3m.dartagnan.program.event.EventFactory.*;
+import static com.dat3m.dartagnan.program.event.EventFactory.OpenCL.newOpenCLBarrier;
+
+public class VisitorOpenCL extends VisitorBase {
```

Comment:
Do we really need a new visitor? At first glance the only difference with the c11 one is the visitor method for the control barrier and how events are tagged. I feel those could be moved to the c11 one, or at least make this visitor extend the c11 one and thus avoiding having the reimplement some of the event visitors

Reply:
Moved into C11 visitor.